src/lib/hostapd.rb

```diff
@@ -9,776 +9,181 @@
 
 module Wifi
 
+  def self.start(config_path, hostapd_path)
+    networks = network_config config_path
+    # TODO: Is this rubyish?
+    if not networks.any?
+      return false
+    end
+    info "found wifis: #{networks.map { |h| h[:name] }}"
+    config_data = create_config_file networks, config_path
+    write_file hostapd_path, config_data
+    # only for debugging purposes:
+    if File.directory? config_path
+      FileUtils.cp hostapd_path, File.join(config_path, 'hostapd.conf.backup')
+    end
+    true
+  end
+
+  private
   def self.wpa_passphrase(ssid, passphrase)
     if [ssid, passphrase].all? { |value| value.to_s.strip.length > 0 }
       OpenSSL::PKCS5.pbkdf2_hmac_sha1(passphrase, ssid, 4096, 32).unpack("H*").first
     end
   end
 
-  module Logging
-    extend Forwardable
-
-    @@logger = Logger.new(STDOUT)
-
-    def self.logger=(logger)
-      @@logger = logger
-    end
-
-    def_delegators :logger, :debug, :info, :warn, :error, :fatal, :unknown
-
-    def log(msg)
-      logger.log
-    end
-
-    def logger
-      @@logger
-    end
-
-
-    def level
-      logger.instance_variable_get(:@level)
-    end
-
-    def level=(level=Logger::ERROR) # DEBUG=0, INFO=1, WARN=2, ERROR=3, FATAL=4, UNKNOWN=5
-      logger.instance_variable_set(:@level, level)
-    end
+  def self.debug msg
+    Logger.new(STDOUT).log Logger::Severity::DEBUG, msg
   end
 
-  module Console
-    include Logging
-
-    def run(cmd)
-      if level == Logger::DEBUG
-        debug "run: #{cmd}"
-      end
-      system "#{cmd} >/dev/null 2>&1"
-    end
+  def self.info msg
+    Logger.new(STDOUT).log Logger::Severity::INFO, msg
   end
 
-  class IPAddr
-    def netmask
-      _to_string(@mask_addr)
-    end
-
-    def succ_net
-      to_range.last.succ
-    end
-
-    def prefixlen
-      @mask_addr.to_s(2).count('1')
-    end
+  def self.network_config(config_path)
+    debug 'fetching networks'
+    networks = []
+    debug 'Looking for private network at ' + File.join(config_path, 'enabled').to_s
+    if File.exists? File.join(config_path, 'enabled')
+      debug 'Private network found, configuring...'
+      networks << {:name => 'private', :path => config_path}
+    end
+    public_network_path = File.join(config_path, 'guest').to_s
+    debug 'Looking for public network at ' + File.join(public_network_path, 'enabled').to_s
+    if File.exists? File.join(public_network_path, 'enabled')
+      debug 'Public network found, configuring...'
+      networks << {:name => 'public', :path => public_network_path}
+    end
+    networks
+  end
 
-    def to_cidr
-      "#{to_s}/#{prefixlen}"
+  def self.get_physical_interface
```

Comment:
In ruby, we usually don't name our getters and setters with get/set, simply `physical_interface` would be the rubyish name for this one. That's because in ruby you can define setters as `def physical_interface=(foo)` which will work fine with regular assignments, so there is no need for "namespacing" get/set like in other languages